src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -73,6 +75,17 @@ object Utilities {
 
   var ingressUrls = mutable.ListBuffer.empty[String]
 
+  def checkIfGTMOrSegment(ruleId: String): Boolean = {
+    val ruleList = List(
+      Constants.segmentPixelRuleId,
+      Constants.segmentAnalyticsRuleId,
+      Constants.googleTagManagerRuleId,
+      Constants.googleTagManagerPixelRuleId
+    )
+    val ruleExists = ruleList.contains(ruleId)
+    ruleExists
+  }
```

Comment:
This can be re-written as an expression 
```
List(
      Constants.segmentPixelRuleId,
      Constants.segmentAnalyticsRuleId,
      Constants.googleTagManagerRuleId,
      Constants.googleTagManagerPixelRuleId
    ).contains(ruleId)